lib/srv/discovery/discovery.go

```diff
@@ -198,3 +267,44 @@ func (s *Server) Wait() error {
 	}
 	return nil
 }
+
+func (s *Server) getAzureSubscriptions(ctx context.Context, subs []string) ([]string, error) {
+	subscriptionIds := subs
+	if slices.Contains(subs, types.Wildcard) {
+		subsClient, err := s.Clients.GetAzureSubscriptionClient()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		subscriptionIds, err = subsClient.ListSubscriptionIDs(ctx)
+		return subscriptionIds, trace.Wrap(err)
+	}
+
+	return subscriptionIds, nil
+}
+
+func (s *Server) initTeleportResourceWatchers(initNode bool) (err error) {
+
+	if !initNode {
+		return nil
+	}
+	s.nodeWatcher, err = services.NewNodeWatcher(s.ctx, services.NodeWatcherConfig{
+		ResourceWatcherConfig: services.ResourceWatcherConfig{
+			Component: teleport.ComponentDiscovery,
+			Log:       s.Log,
+			Client:    s.AccessPoint,
+		},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// wait for nodeWatcher to complete initialization so the
+	// discovery server doesnt get an empty list of nodes and attempt
+	// to perform an installation on all the nodes that do actually
+	// exist.
+	for initNode && !s.nodeWatcher.IsInitialized() {
+		time.Sleep(1 * time.Second)
```

Comment:
instead of time.Sleep can we check periodically for `s.nodeWatcher.IsInitialized` condition to be fulfilled ?